### PR TITLE
fix: Slider Prefix and Suffix, moving state to parent component, update tests and stories

### PR DESCRIPTION
**Pull request by aaacafe786**
Closes # 

## 📝 Description

- add human readable string functionaility to the slider component
- move value and setValue out of the slider and into the parent
- fix tests
- fix stories

## ⛳️ Current behavior (updates)

currently, the sliders min/max labels are not human readable

## 🚀 New behavior

min/max labels are human readable

### CODE REVIEW

**Review by hi-im-si — CHANGES_REQUESTED** [review]
- Storybook is erroring: https://github.com/Open-Study-College/osc/actions/runs/4282036157/jobs/7455793736#step:6:87
- @aaacafe786 please get into the habit of filling in PR details otherwise it provides the reviewer no context into what they are reviewing and in turn making their review more time consuming which is what we are trying to combat

File: packages/osc-ui/src/components/Slider/Slider.tsx [hunk 1]
```diff
@@ -37,42 +40,62 @@ export interface SliderProps extends ComponentPropsWithRef<typeof SliderPrimitiv
...
         const sliderClasses = classNames('c-slider', modifier);
 
+        const generatehumanReadableValue = (value) => {
+            const pluralizedPrefix = prefix && pluralize(prefix, value);
+            const pluralizedSuffix = suffix && pluralize(suffix, value);
+
+            return (
+                (pluralizedPrefix ?? '') + (value as string) + ' ' + (pluralizedSuffix ?? '') ?? ''
+            );
+        };
```
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ui/src/components/Slider/Slider.tsx:57]: nice, i have updated the function

File: packages/osc-ui/src/components/Slider/Slider.tsx [hunk 2]
```diff
@@ -37,42 +40,62 @@ export interface SliderProps extends ComponentPropsWithRef<typeof SliderPrimitiv
...
+                <div
+                    style={{
+                        display: 'flex',
+                        flexDirection: 'row',
+                        justifyContent: 'space-between',
+                    }}
+                >
+                    <p>{generatehumanReadableValue(props.min)}</p>
+                    <p>{generatehumanReadableValue(props.max)}</p>
+                </div>
```
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ui/src/components/Slider/Slider.tsx:97]: updated to span

**Comment by mattwendzina** [comment]
Tried running this but it can't find the `pluralise` module @aaacafe786. Does this need adding to the `osc-ui` pacakge? I can only see it in `osc-studio`. Not exactly sure how everything gets stitched together but I ran `npm run setup` and it was unable to find the module still.

**Comment by aaacafe786** [comment]
@mattwendzina this should be resolved

**Comment by mattwendzina** [comment]
I would add the changes as a new story. Ie a story that gets an updated value through the set timeout which emulates the value changing on the server as this won't always be necessary. Also it appears that the defaultValues don't appear to be working now or having two value points - see the disabled slider. As a result all the tests are also failing. I think if you move this out to a separate story with a separate template (that incorporates the setTimeout) you may fix some of these issues 👍

**Comment by mattwendzina** [comment]
Looks like it's closer 👍 
1)  The Primary should show the dots as default so seems there's still a regression here.
2) The tests are still all failing
3) Think it's worth adding a description to Has Timeout story to explain what is happening here:
e.g. 
HasTimeout.parameters = {
    docs: {
        description: {
            story: 'Description of what is happening...' ,
        },
    },
};

**Comment by mattwendzina** [comment]
Also, appears that the prefix isn't working.
And I guess maybe the suffix too? Don't know though as it isn't being used. Perhaps worth adding the suffix onto the final story too so that it shows both. 
I think it would probably be worth adding a test for the HasTimeout too? Something like ensuring the value gets updated after the timeout?

**Comment by mattwendzina** [comment]
Sorry, I'd also revert the original story to have the 2 default values as this was the original behaviour. Tbh, looking at it there should be two stories - one that is a single value and one that is a range. I think the Slider figma designs got updated further after I originally did it which is why there may only be the one. 
I guess I could do this after, but for now seems like it should be the same as it was, and then I/you can add in the additional story.

**Comment by aaacafe786** [comment]
> Sorry, I'd also revert the original story to have the 2 default values as this was the original behaviour. Tbh, looking at it there should be two stories - one that is a single value and one that is a range. I think the Slider figma designs got updated further after I originally did it which is why there may only be the one. I guess I could do this after, but for now seems like it should be the same as it was, and then I/you can add in the additional story.

@mattwendzina  done

File: packages/osc-ui/src/components/Slider/Slider.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ui/src/components/Slider/Slider.tsx]: @mattwendzina @hi-im-si looks we don't have to redeclare prefix and suffix in the function parameters, they are destructured in the props

**Comment by aaacafe786** [comment]
> Also, appears that the prefix isn't working. And I guess maybe the suffix too? Don't know though as it isn't being used. Perhaps worth adding the suffix onto the final story too so that it shows both. I think it would probably be worth adding a test for the HasTimeout too? Something like ensuring the value gets updated after the timeout?

@mattwendzina done

**Comment by aaacafe786** [comment]
> HasTimeout.parameters = {
>     docs: {
>         description: {
>             story: 'Description of what is happening...' ,
>         },
>     },
> };

@mattwendzina done

**Comment by aaacafe786** [comment]
> Also, appears that the prefix isn't working. And I guess maybe the suffix too? Don't know though as it isn't being used. Perhaps worth adding the suffix onto the final story too so that it shows both. I think it would probably be worth adding a test for the HasTimeout too? Something like ensuring the value gets updated after the timeout?

@mattwendzina done

**Comment by aaacafe786** [comment]
> Looks like it's closer 👍
> 
> 1. The Primary should show the dots as default so seems there's still a regression here.
> 2. The tests are still all failing
> 3. Think it's worth adding a description to Has Timeout story to explain what is happening here:
>    e.g.
> 
> ```
> HasTimeout.parameters = {
>     docs: {
>         description: {
>             story: 'Description of what is happening...' ,
>         },
>     },
> };
> ```

@mattwendzina done

**Comment by mattwendzina** [comment]
Looking good mate 👍 
A couple of other questions/suggestions.
Questions:
- Do we need this 'pluralize' library? I can't actually see what it's doing. I've put some logs in and `pluralizedSuffix` is the same value as the `suffix` that is being passed to it - ie - pounds. Just wondering whether we really need to pull in a library for this as it's just more weight in the codebase. 
- I can't see how/where pluralized prefix is/would be used? Is there an example of this?

Suggestions:
- I think it would be nice if the plural stuff had it's own story labelled as such, rather than just being used on the disabled story. It just makes things a lot clearer. That way you can add a nice description too that describes the behaviour.

File: packages/osc-ui/src/components/Slider/Slider.stories.tsx [hunk 3]
```diff
@@ -68,3 +95,11 @@ isDisabled.args = {
     state: 'isDisabled',
     step: 10,
 };
+
+hasTimeout.parameters = {
+    docs: {
+        description: {
+            story: 'the state of value is not set until two seconds have passed in real time',
```
Comment (mattwendzina) [inline packages/osc-ui/src/components/Slider/Slider.stories.tsx:102]: Sorry, a couple of things here. Me being a bit anal, which I apologise for lol, can we capitalise the start of the sentence? Maybe just a little extra describing why this behaviour is being emulated? E.g. 'This demonstrates a situation where the value isn't immediately available, but received from a server.' Or something like that?

**Comment by aaacafe786** [comment]
> Looking good mate 👍 A couple of other questions/suggestions. Questions:
> 
> * Do we need this 'pluralize' library? I can't actually see what it's doing. I've put some logs in and `pluralizedSuffix` is the same value as the `suffix` that is being passed to it - ie - pounds. Just wondering whether we really need to pull in a library for this as it's just more weight in the codebase.

- https://bundlephobia.com/package/pluralize@8.0.0, its a very small package and it would be a-lot of work to account for all the variations of plurals, deffo possible, but i'm not sue we'd get it under 5kb

- if you pass in pound, and the the value is greater than one,  it will pluralise the word 

- we also use the same package in osc-studio

> * I can't see how/where pluralized prefix is/would be used? Is there an example of this?
```
    return `${pluralizedPrefix ?? ''}${value} ${pluralizedSuffix ?? ''}`.trim();
```

> 
> Suggestions:
> 
> * I think it would be nice if the plural stuff had it's own story labelled as such, rather than just being used on the disabled story. It just makes things a lot clearer. That way you can add a nice description too that describes the behaviour.

-- will do

File: packages/osc-ui/src/components/Slider/Slider.stories.tsx (hunk 3 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Slider/Slider.stories.tsx:102]: done

**Pull request closed without merging**